core/primitives/src/errors.rs

```diff
@@ -749,6 +749,8 @@ pub enum EpochError {
     IOErr(String),
     /// Given account ID is not a validator in the given epoch ID.
     NotAValidator(AccountId, EpochId),
+    /// Error getting information for a shard
+    ShardingError(String),
```

Comment:
Should this be hidden behind a feature-flag and protocol upgrade?

Reply:
Good idea. However, I think it is not necessary to put it in a feature-flag for the following reasons.
- I think `EpochError` is not used on the protocol level, they are all converted to `chain::Error` at some point (@bowenwang1996 is that true?). 
- The logic that triggers this Error also belongs to internal implementation details instead of protocol changes, so I don't want to put that logic in a feature flag
- The error will not be triggered right now with our current implementation because Simple Nightshade is enabled.